lib/src/action_response.cc

```diff
@@ -252,10 +252,25 @@ lth_jc::JsonContainer ActionResponse::toJSON(R_T response_type) const
                     ACTION_STATUS_NAMES.at(ActionStatus::Unknown));
             }
 
-            if (!output.std_out.empty())
+            // If an execution error exists, report that instead of any results.
+            if (action_metadata.includes(EXECUTION_ERROR)) {
+                auto exec_err = action_metadata.get<std::string>(EXECUTION_ERROR);
+                if (!exec_err.empty()) {
+                    lth_jc::JsonContainer err_obj;
+                    err_obj.set("kind", "puppetlabs.pxp-agent/action-error");
```

Comment:
This could also be `execution-error`, seems consistent with the rest of naming.

One caveat with this change is that new agents with old Orchestrator could - potentially - result in sending a message Orchestrator doesn't understand (the _error key) in response to a pxp-module-puppet run that fails due to https://github.com/puppetlabs/pxp-agent/blob/master/lib/src/external_module.cc#L172-L184.

Reply:
@nicklewis an alternative that would be more backward compatible with Orchestrator is to send an rpc_error_message, and add a kind field to that. It'd violate the [PXP spec](https://github.com/puppetlabs/pcp-specifications/blob/master/pxp/versions/1.0/request_response.md#error-message), but nothing except pxp-agent would care.

Reply:
Invalid JSON on stdout can't really happen, can it? That would only be the case for a broken module. So execution-error seems fine, assuming orchestrator doesn't have any automatic weird handling of that.

Reply:
Thin I think this is ready.